numba/typing/dictdecl.py

```diff
@@ -0,0 +1,30 @@
+"""
+This implements the typing template for `dict()`.
+"""
+from __future__ import absolute_import, print_function
+
+from .. import types, errors
+from .templates import (
+    AbstractTemplate,
+    Registry,
+    signature,
+)
+
+registry = Registry()
+infer = registry.register
+infer_global = registry.register_global
+infer_getattr = registry.register_attr
+
+
+_message_dict_support = """
+Unsupported use of `dict()` with positional or keyword argument(s). \
+The only supported use is `dict()`.
+""".strip()
+
+
+@infer_global(dict)
+class DictBuiltin(AbstractTemplate):
+    def generic(self, args, kws):
+        if args or kws:
+            raise errors.TypingError(_message_dict_support)
```

Comment:
wondering if this should be an `UnsupportedError`?

Reply:
We can do that but we have never done that before.  There are no raising of `UnsupportedError` from typing logic.  Also given the "extra error info" (error_rewrite) says `TypingError` is:

> This is not usually a problem with Numba itself but instead often caused by
the use of unsupported features or an issue in resolving types.

I'd prefer to leave it as it.

Reply:
ok, this seems consistent